cloud/disk_manager/test/acceptance/test_runner/eternal_acceptance_test_runner.py

```diff
@@ -1,29 +1,53 @@
+import argparse
 import contextlib
 import logging
 import math
-
+import re
 from concurrent.futures import ThreadPoolExecutor
+from datetime import timedelta
+
+from cloud.blockstore.pylibs import common
+from cloud.blockstore.pylibs.ycp import YcpWrapper
 from .base_acceptance_test_runner import BaseAcceptanceTestRunner, \
     BaseTestBinaryExecutor
-
+from .cleanup import BaseResourceCleaner
 from .lib import (
     check_ssh_connection,
     size_prettifier,
     Error,
 )
 
-from cloud.blockstore.pylibs import common
-
-
 _logger = logging.getLogger(__file__)
 
 
 class EternalAcceptanceTestBinaryExecutor(BaseTestBinaryExecutor):
     _entity_suffix = 'eternal'
 
 
+class EternalTestCleaner(BaseResourceCleaner):
+    def __init__(self, ycp: YcpWrapper, args: argparse.Namespace):
+        super(EternalTestCleaner, self).__init__(ycp, args)
+        test_type = args.test_type
+        disk_size = size_prettifier(args.disk_size * (1024 ** 3)).lower()
+        disk_blocksize = size_prettifier(args.disk_blocksize).lower()
+        disk_name_pattern = re.compile(
+            fr'^acceptance-test-{test_type}-'
+            fr'{disk_size}-'
+            fr'{disk_blocksize}-[0-9]+$',
+        )
+        self._entity_ttls = self._entity_ttls | {
+            'disk': timedelta(days=5),
+        }
+        self._patterns = {
+            **self._patterns,
+            'disk': [disk_name_pattern],
+        }
+
+
 class EternalAcceptanceTestRunner(BaseAcceptanceTestRunner):
     _test_binary_executor_type = EternalAcceptanceTestBinaryExecutor
+    _cleaner_type = EternalTestCleaner
+    _single_disk_test_ttl = timedelta(days=5)
```

Comment:
Why is ttl duplicated here and on line 39?

Reply:
Because we have to cleanup procedures, for run_test_on_single_disk result and for entities created by the python code. The decision to place cleanup close to disks creation is explained by code locality & the thing that in acceptance-acceptance tests we have several disks, and we would like to clean up only them before the single test is ran, but we want to clean up VM's before a vm is created.